### PR TITLE
Simplify argument of Kotpref.asFlowable extension function

### PR DESCRIPTION
**Pull request by chibatching**
## Issue
- no issue

## Overview (Required)
- This PR is just proposal. Because it contains cons almost the same as its pros.
- Simplify Kotpref.asFlowable function's argument by using PreferenceKey interface introduced from Kotpref 2.4.0
  - Pros
    - We can use asFlowable with very simple I/F, its parameter is only need property.
   - Cons
     - This uses reflection, we have to import kotlin-reflect package.

## Links
-

## Screenshot
Before | After
:--: | :--:
 |

### CODE REVIEW

**Comment by takahirom** [comment]
:eyes:

File: app/src/main/java/io/github/droidkaigi/confsched2018/util/ext/KotprefModelExt.kt [hunk 1]
```diff
@@ -2,22 +2,19 @@ package io.github.droidkaigi.confsched2018.util.ext
...
-    return asFlowable(property, context.getString(key))
-}
-
-@CheckResult fun <T> KotprefModel.asFlowable(property: KProperty0<T>, key: String? = null):
-        Flowable<T> {
+@CheckResult fun <T> KotprefModel.asFlowable(property: KProperty0<T>): Flowable<T> {
     return Flowable.create<T>({ emitter ->
-        val listenKey = key ?: property.name
+        property.isAccessible = true
+        val listenKey = (property.getDelegate() as? PreferenceKey)?.key ?: property.name
```
Comment (takahirom) [inline app/src/main/java/io/github/droidkaigi/confsched2018/util/ext/KotprefModelExt.kt]: 📝

**Comment by takahirom** [comment]
It is big 0.9 MB.. kotlin-reflect 🤔 
I don't know it is good to use reflect.lite. How about this?
https://github.com/Kotlin/kotlinx.reflect.lite

**Comment by chibatching** [comment]
I didn't reflect.lite! 😱 
I'll try it!! 👍

**Comment by chibatching** [comment]
It is difficult to accomplish my goal with kotlin-reflect.
So, I close this PR. 
Thank you!

**Pull request closed without merging**